### PR TITLE
refactor(crypto): Make it clear that the encrypt method is for room events

### PR DESCRIPTION
**Pull request by poljar**

### CODE REVIEW

**Review by jplatte — COMMENTED** [review]
Maybe there should be an extra `_content` suffix too? It's not a whole event that gets passed.

**Comment by poljar** [comment]
Eh maybe, but I think that it's sufficiently clear that only servers create whole events. At least I would hope so, if the person is ever touching the crypto crate.

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 1]
```diff
@@ -682,7 +681,7 @@ impl OlmMachine {
     /// Panics if a group session for the given room wasn't shared beforehand.
     ///
     /// [`encrypt()`]: #method.encrypt
```
Comment (gnunicorn) [inline crates/matrix-sdk-crypto/src/machine.rs:683]: broken doc reference

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 2]
```diff
@@ -651,15 +651,14 @@ impl OlmMachine {
     /// Panics if a group session for the given room wasn't shared beforehand.
     ///
     /// [`share_group_session`]: Self::share_group_session
-    pub async fn encrypt(
```
Comment (gnunicorn) [inline crates/matrix-sdk-crypto/src/machine.rs:654]: to make this a non-breaking change, shouldn't we leave a doc-hidden, marked as deprecated old `encrypt`-alias method, too?

**Review by gnunicorn — APPROVED** [review]
Can we make this non-breaking?

**Comment by poljar** [comment]
> Can we make this non-breaking?

Is there a point to this? All the crates that use this have been updated as well. On the other hand nobody is using this crate directly, they are using the bindings.

File: crates/matrix-sdk-crypto/src/machine.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/machine.rs]: Fixed by https://github.com/matrix-org/matrix-rust-sdk/pull/698/commits/f3045dbf995f309b44fffa3bb67af30804e7ccab.

**Comment by gnunicorn** [comment]
image


Well, I guess not.

**Pull request merged**